src/Apps/Sell/Routes/__tests__/DetailsRoute.jest.tsx

```diff
@@ -0,0 +1,162 @@
+const mockPush = jest.fn()
+const mockReplace = jest.fn()
+let submitMutation: jest.Mock
+
+jest.unmock("react-relay")
+jest.mock("System/Router/useRouter", () => ({
+  useRouter: jest.fn(),
+}))
+jest.mock("Utils/Hooks/useMutation")
+
+const submissionMock: Partial<
+  DetailsRoute_Test_Query$rawResponse["submission"]
+> = {
+  category: "Painting",
+}
+
+beforeEach(() => {
+  mockUseRouter.mockImplementation(() => ({
+    router: {
+      push: mockPush,
+      replace: mockReplace,
+    },
+    match: { location: { pathname: "submissions/submission-id/details" } },
+  }))
+
+  submitMutation = jest.fn(() => ({ catch: () => {} }))
+  ;(useMutation as jest.Mock).mockImplementation(() => {
+    return { submitMutation }
+  })
+})
+
+const { renderWithRelay } = setupTestWrapperTL({
+  Component: (props: any) => {
+    return (
+      <SubmissionRoute submission={props.submission}>
+        <DetailsRoute submission={props.submission} />
+      </SubmissionRoute>
+    )
+  },
+  query: graphql`
+    query DetailsRoute_Test_Query @raw_response_type {
+      submission(id: "submission-id") {
+        ...SubmissionRoute_submission
+        ...DetailsRoute_submission
+      }
+    }
+  `,
+})
+
+describe("DetailsRoute", () => {
+  it("renders the Details step", () => {
+    renderWithRelay({
+      ConsignmentSubmission: () => submissionMock,
+    })
+
+    expect(screen.getByText("Artwork details")).toBeInTheDocument()
+    expect(screen.getByText("Back")).toBeInTheDocument()
+    expect(screen.getByText("Continue")).toBeInTheDocument()
+    expect(screen.getByText("Save & Exit")).toBeInTheDocument()
+  })
+
+  it("initializes the form with the submission data", async () => {
+    renderWithRelay({
+      ConsignmentSubmission: () => submissionMock,
+    })
+
+    await waitFor(() => {
+      expect(screen.getByTestId("year-input")).toHaveValue(
+        '<mock-value-for-field-"year">'
+      )
+      expect(screen.getByTestId("medium-input")).toHaveValue(
+        '<mock-value-for-field-"medium">'
+      )
+    })
+  })
+
+  it("saves the submission & navigates to the next step when Continue button is clicked", async () => {
+    renderWithRelay({
+      ConsignmentSubmission: () => submissionMock,
+    })
+
+    screen.getByText("Continue").click()
+
+    await waitFor(() => {
+      expect(mockPush).toHaveBeenCalledWith(
+        '/sell2/submissions/<mock-value-for-field-"externalId">/details'
+      )
+
+      expect(submitMutation).toHaveBeenCalledWith(
+        expect.objectContaining({
+          variables: {
+            input: {
+              externalId: '<mock-value-for-field-"externalId">',
+              year: '<mock-value-for-field-"year">',
+              category: "PAINTING",
+              medium: '<mock-value-for-field-"medium">',
+            },
+          },
+        })
+      )
+    })
+  })
+
+  it("does not save the submission & navigates to the previous step when the back button is clicked", async () => {
```

Comment:
This is something I'd like to discuss later. Maybe we should also save the submission state when the user clicks on "Back". 🤔

Reply:
Yeah, I think it would make sense to save the state and not start over again.

Reply:
Yeah, I agree that it would make sense to also persist any unsaved changes before navigating back. In that case, we might want to reconsider the design treatment of the back control to also take a button form so we can use a button loading state during the mutation/transition.

Reply:
https://artsyproduct.atlassian.net/browse/ONYX-1072